src/target.rs

```diff
@@ -80,13 +80,55 @@ pub fn child_fuzz<'a, D: 'a>(
         }
         debug!("Map size is: {map_size}");
 
+        // The following code will handle CMPLOG and CMPCOV envs
+        //
+        // In AFL++, the cmplog shared memory id env will always set if
+        // users specify "-c" option to `afl-fuzz` in commandline, no
+        // matter whether it is invoking the cmplog_binary or normal binary. See
+        // https://github.com/AFLplusplus/AFLplusplus/blob/c340a022e2546488c15f85593d0f37e30eeaab3a/src/afl-sharedmem.c#L321
+        //
+        // But only for cmplog binaries, AFL++ will set ___AFL_EINS_ZWEI_POLIZEI___, see
+        // https://github.com/AFLplusplus/AFLplusplus/blob/c340a022e2546488c15f85593d0f37e30eeaab3a/src/afl-fuzz-cmplog.c#L34
+        //
+        // As a result, we check both env vars, and users can pass the same unicornafl
+        // binary as both normal binary and cmplog binary (use `-c 0` in `afl-fuzz` cmdline).
+        //
+        // The `CMPLOG_ENABLED` controls whether the underlined `__libafl_targets_cmplog_instructions`
+        // will take effect.
+        let cmp_policy;
+        let has_cmplog = std::env::var("___AFL_EINS_ZWEI_POLIZEI___").is_ok()
+            && libafl_targets::map_cmplog_shared_memory().is_ok();
+        let has_cmpcov = std::env::var("UNICORN_AFL_CMPCOV").is_ok();
+        if has_cmplog {
+            if has_cmpcov {
+                warn!("CMPLOG and CMPCOV turned on at the same time!");
+                warn!("I'll turn off CMPCOV.");
+            }
+            unsafe {
+                CMPLOG_ENABLED = 1;
```

Comment:
(not in the scope of this PR)

I do think we should avoid these global variables in `libafl_targets`

Reply:
A practical way is do like what libafl_qemu's module system does. The map pointer is retrieved from passed-in map observers, but this only hide the global vars under the hood. Anyway, to real handle this problem, a lot of work in LibAFL still needs to be done (mostly in API design since the previous global var is parameter now)